app/src/main/java/com/github/sdpcoachme/CoachMeApplication.kt

```diff
@@ -16,14 +18,10 @@ import com.google.android.libraries.places.api.Places
 import com.google.firebase.FirebaseApp
 import com.google.firebase.database.ktx.database
 import com.google.firebase.ktx.Firebase
-import androidx.datastore.preferences.core.Preferences
-import androidx.datastore.preferences.*
 
-open class CoachMeApplication : Application() {
+private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "coachme_preferences")
```

Comment:
This is the documented way of doing it: we need to make sure that no more than one DataStore is instantiated per Kotlin file, and thus we need to put it at the top of the file, not in the class. In the class, it might be instantiated each time the class constructor is called.